### PR TITLE
Add testing outline

### PR DESCRIPTION
**Pull request by nini-faroux**
So this is ready for review now @Randy1Burrell 

So there ended up being a good bit in it, I'll keep the next PRs much smaller than this.

I reorganised the modules a bit, I replaced the `src/Canonical` with `triphut/Triphut`, and within that I separated the `types` from the `scripts`, just to have a clearer structure.

I noticed a good bit more duplication / repetition, and a good few unused fields in a number of the types. I added a few helper functions in `Shared` to help with reducing the repetition across the modules, and cleaned up some of the helper functions.

I added the test outline, so PSM is working in this PR - I added most of the scripts we'll need in the spec modules for testing. I also added the first tests, a positive and a few negative tests for `mkNftMinter` from the `ConfigurationNft` - I also added tests for some of the helper functions - these are all run with `cabal test`. 

Also added clear comments (Haddock style) in the configuration module.

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -3,27 +3,58 @@
...
+  parseRecord,
+  parseRecordWithModifiers,
+  readField,
+ )
+import Plutus.V1.Ledger.Bytes (getLedgerBytes)
+import Plutus.V1.Ledger.Crypto (PubKeyHash)
+import Plutus.V1.Ledger.Scripts (ValidatorHash)
+import Plutus.V1.Ledger.Tx (TxId (TxId), TxOutRef (TxOutRef))
+import Plutus.V1.Ledger.Value (TokenName)
+import Triphut.AlwaysSucceed (succeed, succeed1, succeedHash, succeedHash1)
```
Comment (Randy1Burrell) [inline app/Main.hs:27]: Why not have a generic name since this is a DAO and we want to opensource the software so that others can use it for their projects? Can you create an issue to have this updated?

**Review by Randy1Burrell — APPROVED** [review]
@nini-faroux great work. I left a comment for you to create an issue then we can merge

File: app/Main.hs (hunk 1 above)
Comment (nini-faroux) [reply app/Main.hs:27]: Thanks Randy, sure that makes sense, will make an issue of that to care of next week

Comment (nini-faroux) [reply app/Main.hs:27]: Thanks Randy, sure that makes sense, will make an issue of that to care of next week
